pushtx/broadcaster.go

```diff
@@ -171,7 +171,17 @@ func (b *Broadcaster) broadcastHandler(sub *blockntfns.Subscription) {
 		// A new broadcast request was submitted by an external caller.
 		case req := <-b.broadcastReqs:
 			err := b.cfg.Broadcast(req.tx)
-			if err != nil && !IsBroadcastError(err, Mempool) {
+			// The Rebroadcaster is not only used by neutrino but
+			// also with other backends now, so we need to make sure
+			// to map the error to the internal BroadcastError.
+			if err != nil {
+				if _, ok := err.(*BroadcastError); !ok {
+					err = MapBackendError(err)
+				}
+			}
+			if err != nil && !IsBroadcastError(err, Mempool) &&
```

Comment:
Re my suggestion in the comments, we'd then pass in that error detection predicate into `IsBroadcastErrror`.

Reply:
Thanks for your suggestion, I implemented it with cut out. My new design removes the duplicated code, but not sure if supplying a `MapCustomBroadcastError` feels intuitive? Maybe you had something even more elegant in mind?